Source/Core/VideoCommon/Assets/MeshAsset.cpp

```diff
@@ -0,0 +1,597 @@
+  {
+    MeshDataChunk chunk;
+    chunk.transform = mat;
+    const tinygltf::Primitive& primitive = mesh.primitives[primitive_index];
+    if (primitive.indices == -1)
+    {
+      ERROR_LOG_FMT(VIDEO, "Mesh '{}' is expected to have indices but doesn't have any", mesh_file);
+      return false;
+    }
+    chunk.material_name = model.materials[primitive.material].name;
+    const tinygltf::Accessor& index_accessor = model.accessors[primitive.indices];
+    const tinygltf::BufferView& index_buffer_view = model.bufferViews[index_accessor.bufferView];
+    const tinygltf::Buffer& index_buffer = model.buffers[index_buffer_view.buffer];
+    const int index_stride = index_accessor.ByteStride(index_buffer_view);
+    if (index_stride == -1)
+    {
+      ERROR_LOG_FMT(VIDEO, "Mesh '{}' has invalid stride", mesh_file);
+      return false;
+    }
+    chunk.indices = std::make_unique<u16[]>(index_accessor.count);
+    auto index_src = &index_buffer.data[index_accessor.byteOffset + index_buffer_view.byteOffset];
+    for (std::size_t i = 0; i < index_accessor.count; i++)
+    {
+      if (index_accessor.componentType == TINYGLTF_COMPONENT_TYPE_UNSIGNED_SHORT)
+      {
+        const auto index_cast = reinterpret_cast<const u16*>(&index_src[i * index_stride]);
+        chunk.indices[i] = *index_cast;
+      }
+      else if (index_accessor.componentType == TINYGLTF_COMPONENT_TYPE_UNSIGNED_BYTE)
+      {
+        const auto index_cast = reinterpret_cast<const u8*>(&index_src[i * index_stride]);
+        chunk.indices[i] = *index_cast;
+      }
+      else if (index_accessor.componentType == TINYGLTF_COMPONENT_TYPE_UNSIGNED_INT)
+      {
+        // TODO: update Dolphin to support u32 indices
+        const auto index_cast = reinterpret_cast<const u32*>(&index_src[i * index_stride]);
+        chunk.indices[i] = static_cast<u16>(*index_cast);
+      }
+    }
+
+    chunk.num_indices = static_cast<u32>(index_accessor.count);
+
+    if (primitive.mode == TINYGLTF_MODE_TRIANGLES)
+    {
+      chunk.primitive_type = PrimitiveType::Triangles;
+    }
+    else if (primitive.mode == TINYGLTF_MODE_TRIANGLE_STRIP)
+    {
+      chunk.primitive_type = PrimitiveType::TriangleStrip;
+    }
+    else if (primitive.mode == TINYGLTF_MODE_TRIANGLE_FAN)
+    {
+      ERROR_LOG_FMT(VIDEO, "Mesh '{}' requires triangle fan but that is not supported", mesh_file);
+      return false;
+    }
+    else if (primitive.mode == TINYGLTF_MODE_LINE)
+    {
+      chunk.primitive_type = PrimitiveType::Lines;
+    }
+    else if (primitive.mode == TINYGLTF_MODE_POINTS)
+    {
+      chunk.primitive_type = PrimitiveType::Points;
+    }
+
+    chunk.vertex_stride = 0;
+    static std::array<std::string_view, 12> all_names = {
+        "POSITION",   "NORMAL",     "COLOR_0",    "COLOR_1",    "TEXCOORD_0", "TEXCOORD_1",
+        "TEXCOORD_2", "TEXCOORD_3", "TEXCOORD_4", "TEXCOORD_5", "TEXCOORD_6", "TEXCOORD_7",
+    };
+    for (int i = 0; i < all_names.size(); i++)
```

Comment:
```suggestion
    for (size_t i = 0; i < all_names.size(); i++)
```

otherwise this will cause a signed/unsigned comparison warning.